src/ElmFormat.hs

```diff
@@ -363,33 +350,41 @@ logErrorOr fn result =
             fn value *> return True
 
 
-doIt :: (InputConsole f, OutputConsole f, InfoFormatter f, FileStore f, FileWriter f) => ElmVersion -> WhatToDo -> Free f Bool
-doIt elmVersion whatToDo =
+
+doIt :: (InputConsole f, OutputConsole f, InfoFormatter f, FileStore f, FileWriter f) => ElmVersion -> Flags.Config -> WhatToDo -> Free f Bool
+doIt elmVersion config whatToDo =
+    let
+        getVersion fileDetectedElmVersion =
```

Comment:
This feels like it's starting to get messy to me.  I'm thinking that we should probably try removing the `elmVersion` param to `doIt` and pushing it into each of the `WhatToDo` constructors that need it, and then we should also resolve the actual versions for each file (this `getVersion` logic) before calling `doIt`.  What do you think?

Reply:
Ok, done.
Things could be even simpler if we get rid of global auto-detection and just use 0.19.
To be discussed, see my other comment about that.